### PR TITLE
misc devex for Zoe, VOM and governance

### PR DESCRIPTION
**Pull request by turadg**
## Description

Miscellaneous code cleanup from https://github.com/Agoric/agoric-sdk/tree/6853-upgrade-flux for #6853

### Security Considerations

--
### Scaling Considerations

--
### Testing Considerations

CI

### CODE REVIEW

File: packages/zoe/src/contractSupport/durability.js [hunk 1]
```diff
@@ -106,9 +106,10 @@ harden(provideChildBaggage);
  * so whatever values an instance needs from other vats must be saved in the first
  * incarnation and read from baggage in each subsequent.
  *
- * This abstracts that condition so that the contract can convert a dictionary of promises
- * into a dictionary of values during its first prepare (start) and each additional
- * automatically reads from the baggage.
+ * This abstracts that condition so that the contract can convert a dictionary
+ * of thunks into a dictionary of values during its first prepare (start) and
+ * each additional automatically reads from the baggage. (The values are thunks
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contractSupport/durability.js]: Is there a subject missing here? `each additional <something> automatically reads from ...`.

**Review by FUDCo — APPROVED** [review]
Looks good, but some of the type decls in liveslots look to me likely to collide with @warner's refactor of the VOM, https://github.com/Agoric/agoric-sdk/pull/7138, due to land shortly.  You may want to hold off on this until that's in.

File: packages/swingset-liveslots/src/collectionManager.js [hunk 2]
```diff
@@ -48,6 +48,19 @@ function throwNotDurable(value, slotIndex, serializedValue) {
   );
 }
 
+/**
+ *
```
Comment (Chris-Hibbert) [inline packages/swingset-liveslots/src/collectionManager.js]: drop blank line, please.

File: packages/swingset-liveslots/src/collectionManager.js [hunk 3]
```diff
@@ -48,6 +48,19 @@ function throwNotDurable(value, slotIndex, serializedValue) {
...
+ * @param {*} syscall
+ * @param {import('./virtualReferences.js').VirtualReferenceManager} vrm
+ * @param {() => number} allocateExportID
+ * @param {() => number} allocateCollectionID
+ * @param {(val: any) => string | undefined} convertValToSlot
+ * @param {*} convertSlotToVal
```
Comment (Chris-Hibbert) [inline packages/swingset-liveslots/src/collectionManager.js:102]: when is `{*}` better than `unknown`?

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 4]
```diff
@@ -379,12 +380,17 @@ export const makeZCFZygote = async (
...
-            creatorFacet,
-            publicFacet,
-            creatorInvitation,
-          ].every(canBeDurable);
-          if (prepare || allDurable) {
+          const areDurable = objectMap(
+            { creatorFacet, publicFacet, creatorInvitation },
+            canBeDurable,
+          );
+          const allDurable = Object.values(areDurable).every(Boolean);
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contractFacet/zcfZygote.js:387]: why do you prefer this version?  It seems to make two passes through the list (admittedly small) to get the same result.

File: packages/zoe/src/contractFacet/zcfZygote.js [hunk 5]
```diff
@@ -379,12 +380,17 @@ export const makeZCFZygote = async (
...
+          if (prepare) {
+            allDurable ||
+              Fail`values from prepare() must be durable ${areDurable}`;
+          }
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contractFacet/zcfZygote.js:391]: good correction.

File: packages/zoe/src/contractSupport/durability.js (hunk 1 above)
Comment (turadg) [reply packages/zoe/src/contractSupport/durability.js]: each additional `prepare`. I'll clarify

File: packages/swingset-liveslots/src/collectionManager.js (hunk 3 above)
Comment (turadg) [reply packages/swingset-liveslots/src/collectionManager.js:102]: `{*}` is `{any}`. `unknown` is better but some declaration is better than nothing

File: packages/zoe/src/contractFacet/zcfZygote.js (hunk 4 above)
Comment (turadg) [reply packages/zoe/src/contractFacet/zcfZygote.js:387]: the status quo is to just say something wasn't durable. this way it tells you which weren't (by keeping track of `areDurable`). I needed it for debugging and I expect others will benefit from it.

**Pull request merged**